src/Store.ts

```diff
@@ -27,29 +23,28 @@ export class Store<S extends { state: object }, C extends ConstructorMap<C>> {
     this.stores = synapse.stores;
   }
 
-  setState = (updater: Updater<S['state']>, { log = true } = {}) =>
-    new Promise(resolve => {
-      const updates: Partial<S['state']> =
-        typeof updater === 'function' ? updater(this.state) : updater;
+  setState = (updater: Updater<State>) => {
+    const updates: Partial<State> = typeof updater === 'function' ? updater(this.state) : updater;
 
-      this.state = { ...this.state, ...updates };
-      this.synapse.updateState(this.id, this.state, { log });
-
-      return resolve(this.state);
-    });
+    this.state = { ...this.state, ...updates };
+    this.synapse.updateState(this.id, this.state);
+  };
 
-  runAsync = async <T>({ key, work, log = true }: RunAsync<T, S['state']>) => {
+  runAsync = async <T>({ key, work }: RunAsync<T, State>) => {
     const errorKey = `${key}Error`;
     const storeKey = key || 'loading';
 
-    await this.setState({ [storeKey]: true, [errorKey]: null } as S['state'], { log });
+    // @ts-ignore
+    await this.setState({ [storeKey]: true, [errorKey]: null });
```

Comment:
We should remove these `await` keywords in `#runAsync` since setState is no longer returning a promise.